challenges/tests/test_models.py

```diff
@@ -24,6 +25,27 @@ def test_challenge_metas(self):
                 self.assertIn(type(step[1]), [ButtonStep, InputStep, TextStep],
                               msg='Steps in {0} needs have a given type'.format(challenge))
 
+    def test_mark_as_completed(self):
+        for challenge_tuple in CHALLENGES:
+            challenge = get_challenge_factory(challenge_tuple[0])
+
+            self.assertFalse(challenge.mark_as_completed())
+            self.assertFalse(challenge.status == Challenge.COMPLETED)
+
+            def convenience_complete(challenge):
+                completion_fields = [field for field in challenge._meta.get_fields() if
+                                     field.name.endswith('_status')]
```

Comment:
Django actually uses a lot of meta programming. Not sure on how to map step function and status field